### PR TITLE
convert from non-standard java_*_file to chef standard deploy

### PR DESCRIPTION
**Pull request by jcam**
This is a breaking change and will require recipe modifications

### CODE REVIEW

File: libraries/provider_file_deploy.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                FileUtils.chown(nil, @new_resource.group, dir)
+                Chef::Log.debug("#{@new_resource} set group to #{@new_resource.group} for #{dir}")
+              end
+            rescue => e
+              raise Chef::Exceptions::FileNotFound.new("Cannot create directory #{dir}: #{e.message}")
+            end
+          end
+        end
+
+        def purge_old_downloads
```
Comment (andreacampi) [inline libraries/provider_file_deploy.rb]: Unless you intend to share this code with the other provider, it's an odd name for local files.

Comment (jcam) [inline libraries/provider_file_deploy.rb]: mmm. good point... so it extends file... Provider::File::LocalDeploy ?

Comment (andreacampi) [inline libraries/provider_file_deploy.rb]: I don't have a problem with the class name, I was talking about `purge_old_downloads`. Perhaps `purge_old_copies` or just `purge_old_files` ?

Comment (jcam) [inline libraries/provider_file_deploy.rb]: Oh. Good point, since it isn't downloaded.

**Pull request merged**